src/Lecture3.hs

```diff
@@ -125,9 +133,11 @@ data Reward = Reward
     } deriving (Show, Eq)
 
 instance Semigroup Reward where
+  Reward g1 s1 <> Reward g2 s2 = Reward (g1 <> g2) (s1 || s2)
 
 
 instance Monoid Reward where
+  mempty = Reward (Gold 0) False
```

Comment:
You can use `mempty` for `Gold` here because it already has a `Monoid` instance 🏅 
```suggestion
  mempty = Reward mempty False
```